db/generateSeedScript.js

```diff
@@ -0,0 +1,111 @@
+
+// style adjustment pricing 
+var priceAdjust = [-20, 20, -10, -10, -10, -10, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 10, 10, 10, 10, 10, 20, 20]
+
+var sizes = {};
+sizes["Men's Shoe"] = ['6','6.5','7','7.5','8','8.5','9','9.5','10','10.5','11','11.5','12','12.5','13','14','15'];
+sizes["Women's Shoe"]  = ['5','5.5','6','6.5','7','7.5','8','8.5','9','9.5','10','10.5','11','11.5','12'];
+
+var chars = ['A','B','C','D','E','F','G','H','I','J','K','L','M','N','O','P','Q','R','S','T','U','V','W','X','Y','Z']
+
+var inventory = [0,0,0,0,0,0,0,0,0,0,1,1,2,2,3,3,4,4,5,5,6,6,7,7,8,8,9,9,10,10]
+
+// a more flexible randomizer that takes arrays and numbers
+var rnd = n => Array.isArray(n) ? n[ rnd(n.length) - 1] : Math.floor( Math.random() * n ) + 1
+
+// generate a random style code ( "AB-1234-styleId" )
+var getStyleCode = id => rnd(chars) + rnd(chars) + (rnd(9000) + 1000) + "-" + id;
+
+// from the base price of a product, returns an array containing plausible base & sale prices for a style 
+var getStylePrice = base => {
+  var adjust = base + rnd( priceAdjust ); // set the adjusted price for this style
+  return rnd(2) == 2 
+          ? [adjust, adjust] // 50% chance of a sale
+          : [adjust, adjust - rnd( adjust / 4 ) - .03] // up to 25% off; sale prices end in .97
+}
+
+// for each of the 100 products, give it a range of style options
+var generateSeedScript = (min, max) => {
+  var styleId = 0, productList = '\n-- products\n', styleList = '\n-- styles\n', inventoryList = '\n-- inventory\n';
+
+  products.forEach( (product, id) => {
+    var productURI = encodeURIComponent( product.toLowerCase().split( ' ' ).join( '-' ) );
+    productList += `INSERT into products (name, nameURL) VALUES ("${product}", "${productURI}");\n`;
+    var stylesForThisProduct = [],
+        numberOfStyles = rnd( max - min ) + min + 1,
```

Comment:
Let's see,
rnd( max - min) // will become
rnd(10 - 2) // which becomes
rnd(8) // this will then give us a number between 1 - 8.

so now,
(any number between 1 -8 ) + min + 1 // will become
(any number between 1 -8 ) + 2 + 1 // will become
(any number between 1 -8 ) + 3 // this will then give us a number between 4 and 11, as opposed to 2 and 10.

i think you meant 
rnd( max - min + 1 ) + min - 1 ?

Reply:
Exactly right. I've updated the generator and results accordingly. Because there are so many inventory records per style, this change ended up pulling almost 3000 records from the seed script!

Reply:
wow!